File: backends/go-gin/internal/domain/client/value_objects.go
package client

import "time"

// ListItem はクライアント一覧レスポンス用の値オブジェクトです。
type ListItem struct {
	ID        uint64
	Name      string
	Status    int
	StartAt   *time.Time
	StopAt    *time.Time
	CreatedAt time.Time
	UpdatedAt time.Time
}

// DetailVo はクライアント詳細レスポンス用の値オブジェクトです。
type DetailVo struct {
	ID          uint64
	Name        string
	Identifier  string
	PostCode    string
	Pref        string
	City        string
	Address     string
	Building    string
	Tel         string
	Email       string
	Status      int
	StartAt     *time.Time
	StopAt      *time.Time
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

// StoreVo はクライアント登録結果の値オブジェクトです。
// メール送信・通知配信に必要なフィールドを含みます。
type StoreVo struct {
	ID          uint64
	Name        string
	Email       string
	AccessToken string
}

// QrVo はQRコードデータのレスポンス用値オブジェクトです。
type QrVo struct {
	Identifier  string
	DeeplinkURL string
}

// InfoVo はスマホアプリ向けクライアント情報のレスポンス用値オブジェクトです。
type InfoVo struct {
	Identifier string
	Name       string
	Status     int
}

// StartVo は利用開始レスポンス用値オブジェクトです。
type StartVo struct {
	AccessToken string
}

File: backends/ruby-rails/app/domain/client/value_objects.rb
# frozen_string_literal: true
#
# クライアントの値オブジェクトを定義するドメインオブジェクトモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Domain
  module Client
    # クライアント一覧の表示用値オブジェクトです。
    ListItem = Struct.new(:id, :name, :status, :start_at, :stop_at, :created_at, :updated_at, keyword_init: true)

    # クライアント詳細の値オブジェクトです。
    DetailVo = Struct.new(
      :id, :name, :identifier,
      :post_code, :pref, :city, :address, :building, :tel, :email,
      :status, :start_at, :stop_at, :created_at, :updated_at,
      keyword_init: true
    )

    # クライアント登録結果の値オブジェクトです。
    StoreResultVo = Struct.new(:id, :name, :email, :access_token, keyword_init: true)

    # QRコードデータの値オブジェクトです。
    QrVo = Struct.new(:identifier, :deeplink_url, keyword_init: true)

    # クライアント情報（スマホアプリ向け）の値オブジェクトです。
    InfoVo = Struct.new(:identifier, :name, :status, keyword_init: true)

    # 利用開始結果の値オブジェクトです。
    StartVo = Struct.new(:access_token, keyword_init: true)
  end
end
